torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -76,10 +70,6 @@ std::vector<at::Tensor> unpack(at::TensorList tl, const char *name, int pos) {
     if (!t.defined()) {
       continue;
     }
-    if (!t.is_variable()) {
-      AT_ERROR("Expected object of type Variable but found type ", t.type().toString(), " at position #", i, " "
-                    "for iterable argument #", pos, " '", name, "'");
-    }
```

Comment:
I believe it's okay to remove the checks in this file, based on the assumption that we never remove the `VariableTensorId` from the tensor type set prior to entering variable type dispatch (and maybe we already have a test for this assumption elsewhere?)

Reply:
Yes, I agree with this claim. We don't have a global assert for this though.